src/python_testing/matter_testing_support.py

```diff
@@ -1492,19 +1492,35 @@ def parse_matter_test_args(argv: Optional[List[str]] = None) -> MatterTestConfig
 
     return convert_args_to_matter_config(parser.parse_known_args(argv)[0])
 
+def async_to_sync(awaitable):
+    "Call sync, the async function."
+    # Usage of new implementation of ChipDeviceController cause the necessary to not use asyncio.run,
+    # because the initialization of `asyncio.Lock` use the `asyncio.get_event_loop()` on init.
+    # Usage of `asyncio.run()` and `asyncio.get_event_loop()` are exclusive.
+    loop = asyncio.get_event_loop()
```

Comment:
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop says "deprecated since version 3.12" and documentation recommends to use alternatives.

I also am not very clear on what is going on based on the above comment and worry about the deprecation and warnings in the python documentation. Would it be correct to use get_running_loop for example?

Or is this code really needing to handle both coroutine vs non-coroutine use cases?

Reply:
Removed change from the PR, the issue is visible on specific version of Python (Python 3.9.x and older), mentioned here https://github.com/project-chip/connectedhomeip/pull/34231#discussion_r1668837629.